src/anomalib/models/image/vlm_ad/lightning_model.py

```diff
@@ -0,0 +1,115 @@
+"""Visual Anomaly Model for Zero/Few-Shot Anomaly Classification."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+
+import torch
+from torch.utils.data import DataLoader
+
+from anomalib import LearningType
+from anomalib.models import AnomalyModule
+
+from .backends import Backend, ChatGPT, Huggingface, Ollama
+from .utils import ModelName, Prompt
+
+logger = logging.getLogger(__name__)
+
+
+class VlmAd(AnomalyModule):
+    """Visual anomaly model."""
+
+    def __init__(
+        self,
+        model: ModelName | str = ModelName.LLAMA_OLLAMA,
```

Comment:
Should this be `model_name` as well? or should it stay as `model`?

Reply:
Depends how we want it in the yaml file.

```yaml
class_path: VlmAd
init_args:
    model: gpt_4o
```

or

```yaml
class_path: VlmAd
init_args:
    model_name: gpt_4o
```

Reply:
Good point. alright, let's leave it as model then